File: api/src/classes/units/Vehicle.js
const Unit = require('../unit')

class Vehicle extends Unit {
  constructor(square, team) {
    super(square, team)
    this.category = 'artillery'
    this.validCommands = ['N', 'E', 'S', 'W', 'C']
  }

  move() {
    super()
    if(this.hasMoved && !this.square.terrain.isObstructed)
      this.setNextCommand(this.command)
  }
}

module.exports = Vehicle

File: api/lib/units/vehicle.rb
require_relative './unit.rb'

class Vehicle < Unit
  def initialize square, team
    super
    @category = :vehicle
    @valid_commands = [:N, :E, :S, :W, :C]

  # vehicles continue rolling unless move failed or moved to obstructed terrain
  def move
    super
    @next_command = @command if @moved? && !@square.terrain.obstructed?
  end
end